### ISSUE TITLE
This is all my mistake. Time to reset.

### ISSUE THREAD

**Issue by vassudanagunta** [upend/IF_MS_BUYS_GITHUB_IMMA_OUT#114]
I inadvertently embarked on [a fool's errand](https://github.com/upend/IF_MS_BUYS_GITHUB_IMMA_OUT/issues/108#issuecomment-395339838). @AnaRobynn's [comment](https://github.com/upend/IF_MS_BUYS_GITHUB_IMMA_OUT/issues/91#issuecomment-395043368) captures the conundrum perfectly. He wants (demands) all of the following:

- **results** *(YES!)*
- **no more Microsoft bashing** *(YES! I wanted to shut [this bashing](https://github.com/upend/IF_MS_BUYS_GITHUB_IMMA_OUT/issues/38) down and would have required more censorship on my part, the very thing he's opposed to.)*
- **a free-speech forum**, allowing anyone to disrupt, jam with noise, or otherwise undermine the work, to usurp this online meeting place for their own agenda *(NOT POSSIBLE if you want results and no bashing).* 
- **results within 4 days** *(even more NOT POSSIBLE if you want all of the above).*

Ironically @AnaRobynn's participation exemplifies this. He contributed [8 words](https://github.com/upend/IF_MS_BUYS_GITHUB_IMMA_OUT/issues/44#issuecomment-395046092) towards "significant results" and [243 words](https://github.com/upend/IF_MS_BUYS_GITHUB_IMMA_OUT/issues/91#issuecomment-395043368) towards "bashing" this effort. 

As @bakkerme [wrote](https://github.com/upend/IF_MS_BUYS_GITHUB_IMMA_OUT/issues/106#issuecomment-395353066) (my emphasis):

> Over the weekend I'm going to look at compiling some general resources about moving to other git hosting providers. **This repo has too much attention to be wasted on bickering in the issues.**

But again, this is all my fault. I should have known better. Instead I tried to compensate for the above contradictions by employing a heavy-hand on the delete and edit buttons. It was a misguided effort to nip in the bud disruptions and noise, to not let this turn into a platform for detractors, saboteurs, shills, to not let such behavior go without consequence, to not reward such behavior by letting their posts stay visible on a highly visible repo. I didn't like doing it, but felt I had no choice. But I did have another choice. And I'm going to make it now. 

### the original intent 

I created this repo when the acquisition was only a rumor for people and projects who count on an independent GitHub (It's right there in the repo name), to make them aware that the independence was under threat, and to enable them to voice their position via the star button in the slim hope that we might influence the decision.  

When rumor became reality, but the number of new stars kept accelerating, using the repo to host a solutions effort seemed to make sense. It all happened so fast, and I felt I had to make a quick decision, without doing the things I should have done first (like making the rules very clear).

**It was never meant to be a public forum** to debate the acquisition. But some people assume that just because there's a text box with a `post` button, it implies a free speech public forum, regardless of what's stated in the README or [in the Issue Template](https://github.com/upend/IF_MS_BUYS_GITHUB_IMMA_OUT/blob/master/.github/ISSUE_TEMPLATE.md). 

I regret my quick decision now.

### git reset HEAD^

To get things back on track, I've decided to restore this repo back to that original intent. 

1. **Instead of hosting projects here, it will link to them.** 
   - **You all can self-organize** and create the kind of projects or forums you want, with whatever rules you want *(You've always had this freedom)*.
   - **To those of you who have criticized how I've run things**: This is your chance to show me. I'll expect you to accept all comers, including me, to let all posts to stand undeleted, or whatever your expectations of me were. I can link to your project with the words "Open forum, all opinions, including pro-Microsoft, welcome."
2. **Issues will be turned off June 8, 9pm Pacific Time, about 30 hours from this posting.** At that point they and comments will no longer be visible.
   - This should give you enough time to share contact information with those you want to team up with, and to export the work you have done here.
   - If anyone has good reason for needing more time, email me and we'll work something out.
3. **I'll post your project's updates here**. Just email me. Or:
4. **I'll accept patches to this repo itself** the pure git way. Without reliance on GitHub. The GitHub version of the repo will simply be a mirror of a repo not dependent on GitHub. This is also in keeping with our goals.

## why this is best

1. It is in keeping with the goals: Get off GitHub.
2. It aligns with a good message: Decentralize.
3. It eliminates the risk of deletion by Microsoft GitHub (You can't `git pull origin` the issues database). The README will eventually get moved to upend.org as well.
4. Many of us had hopes that people would coalesce into teams and we’d rapidly have something to offer. This is the best way to achieve that.
5. Everyone gets what they want, though they don't get it for free anymore. People get the freedom to work on a team with operating principles aligned with their desires (e.g. total free speech vs tight ship)
6. No more dictators (e.g. me). At least not a single dictator.
7. I get to stop deleting posts.
8. I get a break and a decent amount of sleep.

I'll answer any questions or entertain other ideas you may have below.

If you want to debate free speech versus effective organization, I'll be happy to do that. But not here. Feel free to invite me wherever you want to have it.

**Comment by evyatarmeged** [comment]
Seems legit. Let's see where this goes.

**Comment by bakkerme** [comment]
Wow, wasn't expecting it to go nuclear. Sounds like a good chance to learn how to create and apply patch files. I'll still be continuing with my plans, and I hope others who want to help out do so too.

**Comment by xloem** [comment]
I’ve noticed that https://scuttlebutt.nz/ can be good for these situations.  In scuttlebutt, you inherit the streams of the people you follow, so noisemakers fade fast because nobody chooses to follow them.  git-ssb has been mentioned elsewhere.

Another option would be to invite trustworthy passionate community members to aid in managing the noise.  The most effective leaders are those who teach more teachers; they have a light workload but a large impact.

**Comment by Serphentas** [comment]
Since the comparison of alternatives is being done [ħere](https://github.com/bderiso/Microsoft-Github-acquisition) and [here](https://docs.google.com/spreadsheets/d/1LZryaOx4VYHQ_oWLtRjvFrr4RhVjR484eFe7du3Usa4), I would suggest that people really interested in this movement to continue their discussions and debates over there (at least the technical part).

It would be too sad and shameful to lose all what's been done up until now.

**Comment by vassudanagunta** [comment]
@Serphentas Why would anything get lost? I'll post links from here to individual efforts. Best of both worlds: You get to manage it how you want AND the visitors to this repo will be informed about it.

**Comment by xloem** [comment]
I have created a post-github discussion room for networking at https://matrix.to/#/%23postgithub:disroot.org .

Projects like this one generally do get lost.

**Comment by vassudanagunta** [comment]
Just learned we were [mentioned and linked to in a Wired article](https://www.wired.com/story/microsoft-github-code-moderation/).

**Comment by neko** [comment]
Thank you. This is the mature route.

**Comment by vassudanagunta** [comment]
> This is the mature route.

That's a bit condescending @neko (Do you know how old I am and how many projects I've run?), but that don't matter. Show me what you accomplish in the project you lead 😉

**Comment by Krotera** [comment]
>That's a bit condescending @neko (You weren't in my shoes. And do you know how old I am and how many projects I've run?), but that don't matter. Show me what you accomplish in the project you lead wink

Thank you. This is the mature route.

**Comment by AnaRobynn** [comment]
1) results (yes)
2) no more Microsoft bashing (yes/no)
The repo could have taken a more neutral stance on the subject, in stead of trying to create hype by spreading lies.
If people want to bash the repo in the issues go ahead. If people want to bash microsoft go ahead. Issues will be closed since they aren't relevant.
3) See above, just close issues.
4) Not possible to have progress in 4 days? 
Yes is is possible and people have done it already: https://github.com/bderiso/Microsoft-Github-acquisition

5) I want to help the cause succeed, but I lost interest because how things are run here. So yes, I didn't help the cause itself, but rather found it more valuable to give a perspective on why people don't like the movement or lose interest in joining the discussions.

As a general tip, really take note of https://github.com/bderiso/Microsoft-Github-acquisition.
Their tone is a lot more nuanced and for someone having years of experience working in teams you really lack proper communication skills. 

You could have written this entire post in a less "I know I'm right, but do whatever you like and you'll see after that I was right along"-tone. It really doesn't help with the credibility.

Anyway, I'm glad, even though I kinda feels like a shitpost towards me, that my comment moved something and that the repo will become a haven for other repositories (since this one already has the exposure)

**Comment by vassudanagunta** [comment]
Your comment moved something? Wow, some ego. I made these changes because of feedback from other people, and because we all are not of one mind, and that's far uglier than my deleting the comments of "shitposters" to use your favorite term. 

I used your comment because it expressed the contradictory demands some of you are making perfectly.

**Comment by vassudanagunta** [comment]
And yeah, to all of you giving me the thumbs down above and no doubt on this comment too, I'm tired and worn out, because all I've been getting is shit for trying my best. The only reason I haven't deleted this repo is because i feel responsibility towards the 4000+ people who've starred it, and the thousands of visits it continues to get every day.

**Comment by bakkerme** [comment]
Vassudanagunta, you're definitely going through a great deal trying to keep this group under control and your effort is appreciated, but please take it easy. Tensions are high from the acquisition, and understandably so.
If you need any help in establishing the future of the project, please let me know and I'll do what I can to help keep things going.

**Comment by Addvilz** [comment]
It is perfectly obvious by now there is way too much drama involved in all of this "initiative", not to mention this whole thing just smells of personal promotion and some twisted sort of rage activism, fueled by emotions and dissent towards parties involved.

Users not satisfied about GitHub's acquisition should have a resource to be able to consider alternatives, something that would provide ability to make an informed choice. [Such resource as we know has already been created and compiled.
](https://docs.google.com/spreadsheets/d/1LZryaOx4VYHQ_oWLtRjvFrr4RhVjR484eFe7du3Usa4/edit#gid=0). Not sure why this repository exists if we have this.

Both GitHub and Microsoft has failed spectacularly over the years, in many curious ways. Yet, we are still hosting our stuff here, developing stuff here. We are still gaming on Windows. We are still mostly doing graphics and video editing on Windows (please, don't kid yourself by bringing Apple in to this). As of late, Microsoft seems to be embracing open source a lot, you have thank GitHub about it, at least partially. I have to perfectly honest, I am not a huge fan of Microsoft either. Yet none here seem to have considered, that all things given, this might in fact be potentially a good change.

Maybe some GitHub culture and influence towards Microsoft might not be such a bad idea after all?... Maybe, just maybe, the "Evil Corp" could become a little bit less "evil corp" because of this acquisition and the community it will bring? Consider that.

**Comment by ChrisCates** [comment]
@Addvilz, I would be wary of making such an assumption without knowing all of the things Microsoft has been involved in the past, and what they may potentially be involved in now.

To say the least, it is like implying Bill Gates' humanitarian gimmicks should not be taken facetiously for his techno-feudalist desires.

**Comment by Droppers** [comment]
Why don't you start by documenting alternatives with their pros and cons?

**Comment by ChrisCates** [comment]
@Droppers the problem does not rely on pros and cons of new solutions.

The problem relies on the fact that millions of developers have trusted their code with Github and now it is the property of Microsoft.

The implications of this are large, for good, or for bad.

**Comment by vassudanagunta** [comment]
Thanks @bakkerme. Very much. You've been kind and supportive. I'm going to hit you up on that.

One more thing and I'll be done with my ranting and raving. On the subject of "Showing empathy towards other community members", what I care far more about than this GitHub-Microsoft shit are things like racism and sexism. Sexism is especially bad in the tech community because it is so male dominated.

When @mtwentyman pointed out (#58) the possible insensitivity of an image I created to add some humor to this endeavor, I took it seriously (I actually dropped out of tech to work on social justice issues) and asked 9 women I respect what they thought. All but one of them said that it was insensitive to women's pain if not sexist. They didn't explain, because they know me and knew I knew the answer and just needed to open my eyes more. I opened my heart and then it became clear. So I [posted a mea culpa](https://twitter.com/UpEnd_org/status/1004098207494877185) on Twitter where that image has been the most popular thing I've posted. Guess what. Crickets. That makes me sad. And the hearts and the retweets for the original image keep coming in. And look at all the thumbs down @mtwentyman got. He's gotten even more thumbs down than me. Look at that whole thread (#58). I'm pretty disgusted by it. Maybe getting lots of thumbs down in this community is a good sign. 

I'd rather go back to working on social issues and working with underserved children than in this entitled self-important, over-privileged, insensitive male dominated tech industry. If I do continue with the UpEnd project, it will be with a team of people who share the values I have, which are harshly critical of the culture that focuses on wealth and turns a blind eye on those left behind, because, you know, America and especially the tech world is a so-called-meritocracy, i.e. if you're poor or oppressed it's your own fucking fault.

I care about upending all of the over-dominant and oppressive power structures, be it the tech oligarchy of companies like Microsoft, or patriarchy or white supremacy. I just wanted to make that clear. 

"smells of personal promotion and some twisted sort of rage activism" my ass. 

Ok, i'm done. I got it all out. No more from me.

**Comment by ChrisCates** [comment]
@vassudanagunta These are not relevant topics. We are talking about a techno-feudalist overlord looking to grab as much power as it can and is trying to suffocate free software... again...

Also, despite the irrelevant statements, I will interject... But I suggest keeping things on topic.

In regards to your feelings towards feminism, you are focusing on microaggressions. A lot of the world does not have a basic understanding of proto-feminism. This is a far greater issue than the one you posed. Instead of worrying about people's reactions to a meme... Perhaps you should look into reforming countries with Sharia Law. Where it is not uncommon for females to be treated as slaves.

**Comment by ChrisCates** [comment]
I suggest instead of policing yourself (and others) on such minor incidents to reflect on what is happening at a macro scale. The number of injustices that occur daily is tantamount. To worry about a meme is pitiful when there are others in the world who are being stoned to death and oppressed.

**Comment by vassudanagunta** [comment]
Yes, rather than look in the mirror and correct our own behavior, let's deflect by pointing fingers at people that are worse than us.

**Comment by ChrisCates** [comment]
You are literally policing yourself (and others) on a meme...

**Comment by ChrisCates** [comment]
I suggest focusing on getting The Western World caught up to speed on proto-feminism before pursuing such "problems". Not only is this kind of behavior unproductive, it counters the values you wish to achieve in society.

**Comment by vassudanagunta** [comment]
And that's what we should all be doing. People have to live their convictions. Our society has problems because everyone makes excuses. 

And this is on topic. If you read the end of that mea culpa I explain that.

**Comment by vassudanagunta** [comment]
@Addvilz you have no business being here. You said so yourself as you defended sexism in that thread:

> even considering I do not agree with the premise of this repository in general. If you feel personally offended by something, it is your personal responsibility, not anyone elses.

Yeah, women, people of color... they shouldn't be offended. It's their personal responsibility. The white male bubble of the tech world at its best.

**Comment by vassudanagunta** [comment]
I'm deleting the totally off topic and ridiculous exchange between me and @ChrisCates, to spare you all. I'm keeping my rant because it's relevant to the philosophical misalignment behind a lot of our strife, and frankly, it's everyone against me and i've worked so hard on this shit and i'm just going to grant that pass for me. All these comments are gone in 18 hours anyway.

**Comment by vassudanagunta** [comment]
let me try again: no more from me unless it's to answer a question about the OP.

**Comment by Giancarlos** [comment]
I'm under the heavy impression the "trending" repositories and developers are based on traffic at the time, which explains why the CEO of Microsoft was trending with 0 repositories or any user content on his profile. People are just looking up everyone involved with Microsoft. I don't think it's censorship, lots of people came here from HN / 4chan / clown porn sites / whatever and then after a day that attention died down. Which makes the most sense to me. I highly doubt GitHub cares to do censorship on this, it's never truly done that before. Some repositories are just only popular at the moment they receive a ton of attention.

**Comment by vassudanagunta** [comment]
@Giancarlos feel free to repost your theory under #88, but only after you explain the data I posted there.

**Comment by williamknauss** [comment]
Hi @vassudanagunta - Hey mate, I would appreciate a link to my open forum https://github.com/WateringBooth/Refreshment which has a goal to educate people on the many options, and availabilities of git.

**Comment by vassudanagunta** [comment]
## About 7 hours left before this (Issues) is shut down.

**Comment by vassudanagunta** [comment]
@williamknauss 

1. please [see this](https://github.com/upend/IF_MS_BUYS_GITHUB_IMMA_OUT/issues/91#issuecomment-395672150). 

2. presenting your repo as a fork of this one is wrong and disingenuous. 

3. You want to take advantage of the emotion that this repo tapped into *with* its opinionated values while dissing it at the same time?

**Comment by noffle** [comment]
@vassudanagunta I'd to share another viewpoint:

Despite intentions, this repo seems to me more like a public square than a specific person's personal project. It's become a rallying point for people interested in organizing toward making change. Not everyone here is interested in positive change, but [there are issues here that very much are](https://github.com/upend/IF_MS_BUYS_GITHUB_IMMA_OUT/issues/44).

Shutting down this repo, regardless of what others would like who are participating here, *feels like a Microsoft thing to do*. What I mean by that is that you're exerting force from the top of a power hierarchy: because you "own" the repo you've decided that you're going to destroy everything that's been happening here. Is that so different from how a corporation or dictator would respond? What if you saw this space *as a commons*? Something that *you* aren't the owner of, but merely the original facilitator?

I appreciate the heck out of you starting this conversation. It's such an important one to have. Please *let it exist* and let it figure out whatever it is wants to become. Losing the coordination that's happened here benefits corps like MS much more than the process we're all struggling to sort out here. :heart:

**Comment by vassudanagunta** [comment]
@noffle thanks for your respectful viewpoint. 

> Shutting down this repo 

The repo is not going away. As I said above, I feel responsible to the 4000+ people who've starred it.  Just the open forum that this Issues database has turned into is going away, **because I never signed up for that nor offered it**. 

> you're going to destroy everything that's been happening here.

So far what's been happening here is a collection of ideas and research. All of that can easily be copied and published as you see fit. In fact, @Serphentas has been doing that from the get-go.

Well the other thing that has been happening here is people who do not have a shared vision are arguing with each other rather than doing what they believe in *in their own independent effort*. This repo can't be a community effort, because we have no consensus community here.  That is the only part getting shut down. It's only been around less than 5 days anyway. 

> dictator?

I shouldn't even entertain this logical fallacy.  If any organization that has a values statement and only accepts members and contributions that follow those values are dictators, then I guess I am. 

I've never seen a single successful open source project run democratically. They all have either a single benovlent dictator or a small number of devs who act like they are a single persons. In fact, the most successful ones are usually a single developer at least through the first version: That's the best of both worlds of being the dictator but not being a dictator. 

I've been part of tiny startups and worked in large corporations running a large teams on multi-million dollar projects. In all of these cases, getting things done and giving everyone equal voice with no controls never co-exist. There is a time and place for each, but you can't have both at the same time and place. I was part of Occupy, and later attended some Occupy  General Assembly meetings. They had a policy of no leader, and no restrictions on speech. Guess what: 2 hours into the three hour meeting the people that called the meeting didn't get to their agenda, because everyone wanted to make a speech, and then object to the speech just given. We spent 30 minutes because someone wanted to have a moment of silence and then someone raised an objection to having someone's religion (the moment of silence) imposed on them. It was a farce. I knew right there the Occupy movement was dead. And guess what, we all never heard from them again. The protests where hugely valuable, but beyond that the org wasn't able to do anything with it. 

Free speech is great for protests, and for the society, because it is necesary for democracy. But freedom is also the ability to form closed groups to develop something without distraction or opposition. All successful projects and movements start out as one person or a small group before being opened up to everyone. Nascent ideas need a safe womb to gestate, then a period of protected childhood, before being thrust into the world and expected to sink or swim. Linux was first developed in solo. 


> Not everyone here is interested in positive change

But I and this repo have been and stated so clearly from the get-go, from the README ("Some believe that the concentration of so much power in the hands of so few is antithetical not just to free and open software, but also to a free and just society. We, the contributors and stargazers of this project are such people."), to the organization's description ("Upend the web oligarchy. #resist"), on the [upend.org](http://www.upend.org) to the website ("a future rebel base of the web resistance. take back the web. take back our world. #resist").

And perhaps most importantly, in all my messages on Twitter to make people aware of the effort that drove a lot of traffic here.  That Twitter traffic was what bootstrapped this repo into the Trending list. I had all of 3 or 4 GitHub followers before this, so it had nothing to do with me. It had everything to do with my almost 24-hour messaging efforts on Twitter, which have been clearly opinionated.

So no, this is not going away, and no, it is not going to become vanilla or "neutral". It's going to stay true to all of the things I listed above.

**Comment by vassudanagunta** [comment]
If I sound opinionated, I am. 

If I sound angry, I am. One is at the injustice in society and how the power structures perpetrate those injustices.

If sound frustrated, I am. I a bit tired of all the free speech grandstanding, while not using that speech against sexism or the injustices in society. It all seems rather selfist to me. 

Checkout what people say about me as a leader on LinkedIn.

**Comment by Apollia** [comment]
Thanks you so much for all your wonderful efforts!

In my opinion, moderating a repo or other forum to keep it true to its intended purpose is just keeping it on topic, not censorship.

Anyone who wants complete freedom of speech can have that elsewhere.


<br>Total freedom of speech too often just turns into vicious chaos.

Obvious example: unmoderated Usenet groups.  https://en.wikipedia.org/wiki/Usenet

Even somewhat moderated forums can easily turn very nasty, even just because of one or a few bad-intentioned people.  Or even as a result of well-intentioned people communicating in a way that is perceived as rude or offensive (even if those people didn't even mean to be rude or offensive).

Then, many people end up wasting a lot of time arguing instead of actually getting useful things done.


<br>I think many poorly-run forums qualify as more of a wrecknology than a technology.

https://medium.com/bad-words/the-wrecknology-boom-472c1be52d9#.lc9re51b2

(Wrecknology is also a clever, appropriate term for many other technologies.  I'm sure we can all think of some, so no need for me to name any specific ones.)


<br>I think it's pretty much impossible to have a mostly pleasant, on-topic public forum without some amount of moderation.

And trying to moderate a large number of posts is time-consuming and can be quite unpleasant, tedious work, especially if you're just one unpaid volunteer.


<br>So, I totally support your restoration of this repo to its original intent.

Thanks again to you and everyone for all your wonderful efforts!

Partly because of the resources people here have shared, I feel like I'm a lot closer now to figuring out how to set up my own personal GitHub alternative that will enable me to publish my own Git repos on my own websites.

**Comment by Serkan-devel** [comment]
Can't the issue-comments only get locked for no further replies? This could be interesting as a historical event later on

**Comment by vassudanagunta** [comment]
fyi, in case anyone wants to work with this dictator, some things i’d like to do besides give people good advice on their get off github options:

1. a petition for people to sign (by pull request) asking Github to (1) open source it’s code or donate it to a non-profit to host an independent home of open source, since i believe they pulled a bait and switch on us so owe us this and (2) provide a way for people to migrate their issues and such off. 

2. #104 or something similar. need a profile pic or overlay that everyone could use to make ourselves even more visible 

or if someone already had got this, i’ll link to it.

**Comment by vassudanagunta** [comment]
Hello,

The GitHub Evacuation Project has moved to GitLab (not an endorsement or even a final home). Your enthusiasm and contribution is still needed. Please check out [the new project home](https://gitlab.com/upend/github), and read [the project wiki](https://gitlab.com/upend/github/evac-guide/wikis) for info to get restarted.

Thank you!

vas
